crates/ide_assists/src/handlers/convert_while_to_loop.rs

```diff
@@ -0,0 +1,175 @@
+use std::iter::once;
+
+use syntax::{
+    ast::{
+        self,
+        edit::{AstNodeEdit, IndentLevel},
+        make, LoopBodyOwner,
+    },
+    AstNode,
+};
+
+use crate::{
+    assist_context::{AssistContext, Assists},
+    utils::invert_boolean_expression,
+    AssistId, AssistKind,
+};
+
+// Assist: convert_while_to_loop
+//
+// Replace a while with a loop.
+//
+// ```
+// fn main() {
+//     $0while cond {
+//         foo();
+//     }
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     loop {
+//         if !cond {
+//             break;
+//         }
+//         foo();
+//     }
+// }
+// ```
+pub(crate) fn convert_while_to_loop(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let while_expr: ast::WhileExpr = ctx.find_node_at_offset()?;
```

Comment:
```suggestion
    let while_kw = ctx.find_token_syntax_at_offset(T![while])?;
    let while_expr: ast::WhileExpr = while_kw.parent().and_then(ast::WhileExpr::cast)?;
```
The assist should only trigger when the cursor is on the `while` keyword to not pollute the assist menu.